app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,27 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      sucess = false
+      case status
+      when 'publish'
+        success = true if can?(:update, media_object)
+      when 'unpublish'
+        success = true if can?(:unpublish, media_object)
+      end
+      if success
+        success_ids << id
       else
-        case status
-        when 'publish'
-          success_ids << id
-        when 'unpublish'
-          if can? :unpublish, media_object
-            success_ids << id
-          else
-            errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
-          end
-        end
+        errors <<
+          "#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."
       end
     end
-    flash[:success] = t("blacklight.status.success", count: success_ids.count, status: status) if success_ids.count > 0
-    flash[:alert] = "#{t('blacklight.status.alert', count: errors.count, status: status)}</br> #{ errors.join('<br/> ') }".html_safe if errors.count > 0
+    flash[:success] =
+      "#{t("blacklight.status.success", count: success_ids.count, status: status)} "\
+      "#{t('blacklight.status.bulk_warning')}" if success_ids.count > 0
+    flash[:alert] = 
+      "#{t('blacklight.status.alert', count: errors.count, status: status)} "\
+      "</br> #{ errors.join('<br/> ') }".html_safe if errors.count > 0
```

Comment:
Good point. On the plus side, I didn't write the code this is based on. :smile: 
Will sanitize.